app/scripts/metamask-controller.js

```diff
@@ -3041,17 +3031,17 @@ export default class MetamaskController extends EventEmitter {
 
     this.controllerMessenger.subscribe(
       `${this.snapController.name}:snapUninstalled`,
-      (truncatedSnap) => {
-        const notificationIds = Object.values(
-          this.notificationController.state.notifications,
-        ).reduce((idList, notification) => {
-          if (notification.origin === truncatedSnap.id) {
-            idList.push(notification.id);
-          }
-          return idList;
-        }, []);
-
-        this.dismissNotifications(notificationIds);
+      async (truncatedSnap) => {
```

Comment:
This doesn't seem like it needs to be async? I think `deleteNotificationsById` is async by mistake?